metricflow/engine/metricflow_engine.py

```diff
@@ -684,32 +676,59 @@ def simple_dimensions_for_metrics(  # noqa: D102
         )
         return self._filter_linkable_dimensions(linkable_element_set=linkable_element_set)
 
-    def _filter_linkable_dimensions(self, linkable_element_set: LinkableElementSet) -> List[Dimension]:
+    def _filter_linkable_dimensions(self, linkable_element_set: BaseLinkableElementSet) -> List[Dimension]:
         dimensions: List[Dimension] = []
-        for path_key, linkable_dimensions_tuple in linkable_element_set.path_key_to_linkable_dimensions.items():
-            for linkable_dimension in linkable_dimensions_tuple:
-                if LinkableElementProperty.METRIC_TIME in linkable_dimension.properties:
-                    dimension = self._build_metric_time_from_linkable_dimension(linkable_dimension)
+
+        for annotated_spec in linkable_element_set.annotated_specs:
+            properties = annotated_spec.properties
+            element_type = annotated_spec.element_type
+            if element_type is LinkableElementType.TIME_DIMENSION:
+                # Simple dimensions shouldn't show date part items.
```

Comment:
If we're adding a restriction that makes this function only applicable to simple dimensions, we should rename it to make that clear (e.g., `_filter_simple_linkable_dimensions()`)

Reply:
Updated.